File: 1 лаба/C++/A.cpp
#include <iostream>
#include <vector>
using namespace std;

void SelectionSort(vector<int> &arr) {
  int n = arr.size();

  for (int i = 0; i < n - 1; i++) {
    int maxi = i;

    for (int j = i + 1; j < n; j++) {
      if (arr[j] > arr[maxi]) {
        maxi = j;
      }
    }

    int temp = arr[i];
    arr[i] = arr[maxi];
    arr[maxi] = temp;
  }
}

int main() {
  vector<int> arr;
  int x;

  while (cin >> x) {
    arr.push_back(x);
  }

  SelectionSort(arr);

  for (int i = 0; i < (int)arr.size(); i++) {
    cout << arr[i];
    if (i + 1 < (int)arr.size()) {
      cout << " ";
    }
  }
  cout << endl;

  return 0;
}

File: 1 лаба/Python/A.py
def SelectionSort(arr):
    n = len(arr)
    for i in range(n - 1):
        max_i = i
        for j in range(i + 1, n):
            if arr[j] > arr[max_i]:
                max_i = j
        arr[i], arr[max_i] = arr[max_i], arr[i]
    return arr


arr = list(map(int, input().split()))
print(*SelectionSort(arr))
